PMSCore/ReportManager.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using PMSCore;
+
+namespace PMSCore
+{
+    /// <summary>
+    /// This class handles the creation and management of reports.
+    /// </summary>
+    public sealed class ReportManager
+    {
+        // The single instance of ReportManager
+        private static readonly ReportManager _instance = new ReportManager();
+
+        /// <summary>
+        /// Private constructor to prevent external instantiation.
+        /// </summary>
+        private ReportManager()
```

Comment:
should init logger here

Reply:
@DistroRickDev , this should be done now, can you please double-check?